lib/pdc/config.rb

```diff
@@ -107,10 +112,10 @@ def reset_logger
 
       # resets and returns the +Faraday+ +connection+ object
       def reset_base_connection
+        clear_token
         headers = PDC::Request.default_headers
         PDC::Base.connection = Faraday.new(url: api_url, headers: headers) do |c|
           c.request   :append_slash_to_path
-          c.request   :authorization, 'Token', token if config.requires_token
```

Comment:
Instead of removing the middleware, could we write a new middleware that _may be_ inherits `authorization` middleware which fetches token only once?